src/goto-instrument/contracts/assigns.h

```diff
@@ -34,7 +34,7 @@ class assigns_clauset
 
     bool operator==(const targett &other) const
     {
-      return expr == other.expr;
+      return address == other.address;
```

Comment:
This is inconsistent with the hash function, because that hashes the `expr` not `address`.
We don't want to have two targets such that `t1 == t2` but `hash(t1) != hash(t2)`.

Reply:
You're correct. I fixed the hash function as well.